### ISSUE TITLE
Deprecate or remove unused warnings and exceptions in twisted.trial.util

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4505]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)| @cyli reported|
|-|-|
|Trac ID|trac#4505|
|Type|task|
|Created|2010-06-15 08:35:36Z|
|Branch|https://github.com/twisted/twisted/tree/remove-trialutil-exceptions-4505|

FailureError, DirtyReactorWarning, DirtyReactorError, PendingTimedCallsError have been deprecated since Twisted 8.0, although the deprecation was stated in a docstring rather than as a warning.

This should either be deprecated correctly (via warnings) or removed.  Removing them does not seem to affect any tests.

Attachments:

* [4505.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/198/198a1afb887f716187a7b1bef7cda486a3ef6dd8/911e7809cc9d99ec8d6d594c03b477411b132c68.diff) (2135 bytes) - added by cyli on 2010-06-15 08:41:05Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4505 4505
type__task task
reporter__cyli cyli
priority__normal normal
milestone__ 
branch__branches_remove_trialutil_exceptions_4505 branches/remove-trialutil-exceptions-4505
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__easy easy
time__1276590936000000 1276590936000000
changetime__1276746166000000 1276746166000000
version__None None
owner__cyli cyli
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_1"></a>|@cyli removed owner|
|-|-|

I am in favor of removal, as it is simpler.  Also, the docstring for twisted.trial.util say that it is a module for internal use only.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_3"></a>|@jesstess commented|
|-|-|

(In [29309]) Branching to 'remove-trialutil-exceptions-4505'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess commented|
|-|-|

(In [29310]) Apply patch 4505.diff by cyli.

refs [#4505](https://github.com/twisted/twisted/issues/4505)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess set owner to @cyli|
|-|-|

Thanks for tackling all these deprecations and removals, cyli! 

Given the "This code is for Trial's internal use.  Do NOT use this code if you are writing tests.  It is subject to change at the Trial maintainer's whim." disclaimer at the top of util.py, I agree with removal instead of deprecation.

Some feedback:

 * In the topfile, the message is duplicated.
 * In util.py, pyflakes catches that _ _all_ _ still references the removed Exceptions.
 * There are still references to DirtyReactorError in twisted/trial/test/test_reporter.py and twisted/trial/reporter.py

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_6"></a>|@cyli commented|
|-|-|

(In [29313]) Bump copyright (previous commit: fixes for review)

refs [#4505](https://github.com/twisted/twisted/issues/4505)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli commented|
|-|-|

>  * In the topfile, the message is duplicated.

Huh, odd, it wasn't in the diff.  Fixed though.

>  * In util.py, pyflakes catches that _ _all_ _ still references the removed Exceptions.

Removed.

>  * There are still references to DirtyReactorError in twisted/trial/test/test_reporter.py and twisted/trial/reporter.py

Looks like someone forgot to modify the comments when they changed over from using DirtyReactorError to using DirtyReactorAggregateError.  Fixed.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)|@cyli removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess set owner to @cyli|
|-|-|

C{DirtyReactorAggregateError} might as well be L{DirtyReactorAggregateError} in test_warningsEnabled and test_warningsMaskErrors. Other than that, I think this good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_11"></a>|@cyli commented|
|-|-|

(In [29316]) Updated C{DirtyReactorAggregateError} to L{DirtyReactorAggregateError}

refs [#4505](https://github.com/twisted/twisted/issues/4505)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_12"></a>|@cyli set status to `closed`|
|-|-|

(In [29317]) Merge remove-trialutil-exceptions-4505 : remove deprecated exceptions in t.trial.util

Author: cyli
Reviewer: jesstess
Fixes: [#4505](https://github.com/twisted/twisted/issues/4505)

Remove deprecated warnings and exceptions from t.trial.util, as well as some comments in t.trial.reporter and t.trial.test.test_reporter

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_13"></a>|@cyli commented|
|-|-|

Thanks for the quick turnaround time!
